release-tool: fix config-forking

dhiller opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Consider creating this PR as draft: https://github.com/kubevirt/kubevirt/blob/main/CONTRIBUTING.md#consider-opening-your-pull-request-as-draft
2. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
-->

**What this PR does / why we need it**:

config-forking was broken in a couple of ways:
* it used an old version of the config-forker from test-infra which didn't retain the value for `run_before_merge`
* it didn't update the forked config, since for release branches the phase plugin didn't work in the same way, therefore the values had to be converted from `run_before_merge` to `always_run`

This PR fixes the issues.

We also now use this opportunity to remove the bazel cache label since
we need to do that every time anyways: https://github.com/kubevirt/project-infra/pull/3961

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3478

**Special notes for your reviewer**:
/cc @xpivarc @enp0s3

xpivarc commented on releng/release-tool/release-tool.go:
```diff
@@ -467,6 +482,44 @@ func (r *releaseData) forkProwJobs() error {
+	if err != nil {
+		return err
+	}
+	var newPresubmits []config.Presubmit
```
don't you want to define this in the loop?

dhiller replied:
Good catch, thank you!

xpivarc commented on releng/release-tool/release-tool_test.go:
```diff
@@ -457,3 +460,37 @@ func TestNewTag(t *testing.T) {
 	}
 }
+
+func Test_convertRunBeforeMergeToAlwaysRun(t *testing.T) {
```
nit: Without the _

dhiller replied:
As said in another PR, since this is a non-exported function the IDE would not find it. Also it's more clear and less annoying to read (IMHO) - but if you insist...

dhiller replied:
![image](https://github.com/user-attachments/assets/fdc4fd1b-dbe1-4062-8b2b-08bd0969bce3)

^^ this is what I mean

dhiller commented:
@xpivarc updated PR, PTAL, thank you!

brianmcarey commented:
/cc

brianmcarey commented on releng/release-tool/release-tool.go:
```diff
@@ -467,6 +482,48 @@ func (r *releaseData) forkProwJobs() error {
 	return nil
 }
 
+// configureReleaseJob changes the values from the configuration requuired for jobs running against
```
nit typo:
```suggestion
// configureReleaseJob changes the values from the configuration required for jobs running against
```

brianmcarey reviewed (approved):
/approve 

thanks @dhiller 

I am going to place a hold just in case you want to address the very tiny nit I came across. 

/hold

dhiller commented:
> /approve
> 
> thanks @dhiller
> 
> I am going to place a hold just in case you want to address the very tiny nit I came across.
> 
> /hold

@brianmcarey thanks for the review, addressed your comment, PTAL! :smiley_cat:

brianmcarey reviewed (commented):
/lgtm

dhiller commented:
/hold cancel
